jmxtrans-core/src/test/java/com/googlecode/jmxtrans/model/naming/typename/TypeNameValueTest.java

```diff
@@ -0,0 +1,106 @@
+    }
+
+    @Test
+    public void testExtractsMultiple() {
+        assertExtractsTypeNameValues("key1=value1,key2=value2",
+                new TypeNameValue("key1", "value1"),
+                new TypeNameValue("key2", "value2"));
+        // reorder
+        assertExtractsTypeNameValues("key2=value2,key1=value1",
+                new TypeNameValue("key2", "value2"),
+                new TypeNameValue("key1", "value1"));
+    }
+
+    @Test
+    public void testExtractsWithoutValue() {
+        assertExtractsTypeNameValues("key",
+                new TypeNameValue("key"));
+        assertExtractsTypeNameValues("key1,key2",
+                new TypeNameValue("key1"),
+                new TypeNameValue("key2"));
+    }
+
+    @Test
+    public void testExtractsMixedCases() {
+        assertExtractsTypeNameValues("key1=value1,,key2=value2,key3",
+                new TypeNameValue("key1", "value1"),
+                new TypeNameValue("key2", "value2"),
+                new TypeNameValue("key3"));
+        assertExtractsTypeNameValues("key1=value1,key2,,key3=value3",
+                new TypeNameValue("key1", "value1"),
+                new TypeNameValue("key2"),
+                new TypeNameValue("key3", "value3"));
+        assertExtractsTypeNameValues(",key1,key2=value2,key3=value3",
+                new TypeNameValue("key1"),
+                new TypeNameValue("key2", "value2"),
+                new TypeNameValue("key3", "value3"));
+        assertExtractsTypeNameValues("key1,key2=value2,key3,",
+                new TypeNameValue("key1"),
+                new TypeNameValue("key2", "value2"),
+                new TypeNameValue("key3"));
+    }
+
+    @Test
+    public void testExtractMap() {
+        assertThat(TypeNameValue.extractMap(null)).isEmpty();
+        assertThat(TypeNameValue.extractMap("")).isEmpty();
+        assertThat(TypeNameValue.extractMap("x-key1-x")).isEqualTo(makeMap("x-key1-x", ""));
+        assertThat(TypeNameValue.extractMap("x-key1-x,x-key2-x")).isEqualTo(makeMap("x-key1-x", "", "x-key2-x", ""));
+        assertThat(TypeNameValue.extractMap("x-key1-x=x-value1-x")).isEqualTo(makeMap("x-key1-x", "x-value1-x"));
+
+        assertThat(TypeNameValue.extractMap("x-key1-x=x-value1-x,y-key2-y=y-value2-y"))
+                .isEqualTo(makeMap("x-key1-x", "x-value1-x", "y-key2-y", "y-value2-y"));
+        assertThat(TypeNameValue.extractMap("x-key1-x=x-value1-x,y-key2-y=y-value2-y,z-key3-z=z-value3-z"))
+                .isEqualTo(makeMap("x-key1-x", "x-value1-x", "y-key2-y", "y-value2-y", "z-key3-z", "z-value3-z"));
+        assertThat(TypeNameValue.extractMap("x-key1-x=x-value1-x,y-key2-y,yy-key2.5-yy=a=1,z-key3-z=z-value3-z"))
+                .isEqualTo(makeMap("x-key1-x", "x-value1-x", "y-key2-y", "", "yy-key2.5-yy", "a=1", "z-key3-z", "z-value3-z"));
+    }
+
+    /** Convenience method for creating a Map for comparison. */
+    private Map<String, String> makeMap(String... keysAndValues) {
```

Comment:
You could use [ImmutableMap.of()](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/ImmutableMap.html#of(K,%20V,%20K,%20V,%20K,%20V)), which provides almost the same method.